tsconfig.base.json

```diff
@@ -0,0 +1,31 @@
+{
+	"compilerOptions": {
+		"allowJs": true,
+		"checkJs": true,
+		"emitDeclarationOnly": true,
+		"declaration": true,
+		"module": "esnext",
+		"moduleResolution": "node",
+		"target": "es5",
+		"composite": true,
+
+		"strict": true,
+		"strictNullChecks": true,
+		"noImplicitAny": true,
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noImplicitReturns": true,
+		"noFallthroughCasesInSwitch": true,
+
+		/* This needs to be false so our types are possible to consume without setting this */
+		"esModuleInterop": false,
+		"isolatedModules": true
+	},
+	"exclude": [
+		"**/build",
+		"**/dist",
+		"**/__tests__",
+		"**/benchmark",
+		"**/node_modules"
+	]
```

Comment:
Still wonder about whether this is necessary, per https://github.com/aduth/tannin/pull/4#discussion_r396031211 .

Reply:
This may not be necessary. I don't think it does any harm, however. These are sensible defaults to extend from.

With every project using an explicit set of `file` rather than `include` patterns, we could probably drop the exclude.